core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -251,10 +264,22 @@ private void releaseInfrastructure(TestPlan testPlan,
      *
      * @param testPlan TestPlan object to persist
      */
-    private void setTestPlanStatus(TestPlan testPlan) {
+    private void persistTestPlanStatus(TestPlan testPlan) {
         try {
-            Status status = isFailedTestScenariosExist(testPlan) ? Status.FAIL : Status.SUCCESS;
-            setTestPlanStatus(testPlan, status);
+            boolean isSuccess = true;
+            for (TestScenario testScenario : testPlan.getTestScenarios()) {
+                if (testScenario.getStatus() != Status.SUCCESS) {
+                    isSuccess = false;
+                    if (testScenario.getStatus() == Status.FAIL) {
```

Comment:
What about other statuses of testScenarios such as DID_NOT_RUN and ERROR? How about:

```
If there is at least one FAIL, the overall status is  --> FAIL
Else, if there is at least one ERROR, the overall status is  --> ERROR
Else, the overall status is --> SUCCESS.
```
What about 'DID_NOT_RUN' status then? We have to think abt what the overall status should be in that case. The logic in this comment sets the overall status to SUCCESS for that case as well currently.

Reply:
ERROR status is already set when scenario execution fails which is caught in TestPlanExecutor#runScenarioTests. Therefore, at this point, the ERROR status is already set if there exist any failed tests.

So additional checking of failed tests and success would suffice. WDYT?

Reply:
Created #610 to resolve this